### PR TITLE
Evolution Graphics and others

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/evolution_graphics.c [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+        gTasks[taskId].func = EvoTask_PostEvoSparklesSet1Teardown;
+    }
+}
+
+static void EvoTask_PostEvoSparklesSet1Teardown(u8 taskId)
+{
+    DestroyTask(taskId);
+}
+
+u8 LaunchTask_PostEvoSparklesSet2AndFlash(u16 a0)
```
Comment (jiangzhengwenjz) [inline src/evolution_graphics.c]: in `include/evolution_graphics.h`, a0 is species

File: src/evolution_graphics.c [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+    }
+}
+
+static void EvoTask_PostEvoSparklesSet2Teardown(u8 taskId)
+{
+    if (!gPaletteFade.active)
+        DestroyTask(taskId);
+}
+
+u8 LaunchTask_PostEvoSparklesSet2Alt(u16 a0)
```
Comment (jiangzhengwenjz) [inline src/evolution_graphics.c]: in the header there's a function called LaunchTask_PostEvoSparklesSet2AndFlash_Trade, is it the same function?

File: include/evolution_graphics.h [hunk 3]
```diff
@@ -10,7 +10,7 @@ u8 LaunchTask_PreEvoSparklesSet2(void);
 u8 LaunchTask_PostEvoSparklesSet1(void);
 u8 LaunchTask_PostEvoSparklesSet2AndFlash(u16 species);
 u8 LaunchTask_PostEvoSparklesSet2AndFlash_Trade(u16 species);
-u8 sub_817C3A0(u8 preEvoSpriteID, u8 postEvoSpriteID);
+u8 LaunchTask_PrePostEvoMonSprites(u8 preEvoSpriteId, u8 postEvoSpriteId);
 
 #define EvoGraphicsTaskEvoStop  data[8]
```
Comment (jiangzhengwenjz) [inline include/evolution_graphics.h]: does this mean the same as tBButtonCancel?

Comment (PikalaxALT) [reply include/evolution_graphics.h]: yes

**Pull request merged**